drivers/spi/spi_ll_stm32.c

```diff
@@ -121,8 +119,12 @@ static void spi_stm32_shift_s(SPI_TypeDef *spi, struct spi_stm32_data *data)
 			spi_context_update_tx(&data->ctx, 2, 1);
 		}
 	}
+	else
+		LL_SPI_DisableIT_TXE(spi);
+
+	if (LL_SPI_IsActiveFlag_RXNE(spi) && spi_context_rx_on(&data->ctx)) {
+		u16_t rx_frame;
 
-	if (LL_SPI_IsActiveFlag_RXNE(spi)) {
 		if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
 			rx_frame = LL_SPI_ReceiveData8(spi);
 			if (spi_context_rx_on(&data->ctx)) {
```

Comment:
Similarly, the change means that RX on has already been checked here, so this existing check (and the one below) should be removed.

Reply:
OK